x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/StateStreamer.java

```diff
@@ -76,17 +76,21 @@ public void restoreStateToStream(String jobId, ModelSnapshot modelSnapshot, Outp
                     .setSize(1)
                     .setQuery(QueryBuilders.idsQuery().addIds(stateDocId))
                     .get();
-                if (stateResponse.getHits().getHits().length == 0) {
-                    LOGGER.error(
-                        "Expected {} documents for model state for {} snapshot {} but failed to find {}",
-                        modelSnapshot.getSnapshotDocCount(),
-                        jobId,
-                        modelSnapshot.getSnapshotId(),
-                        stateDocId
-                    );
-                    break;
+                try {
+                    if (stateResponse.getHits().getHits().length == 0) {
+                        LOGGER.error(
+                            "Expected {} documents for model state for {} snapshot {} but failed to find {}",
+                            modelSnapshot.getSnapshotDocCount(),
+                            jobId,
+                            modelSnapshot.getSnapshotId(),
+                            stateDocId
+                        );
+                        break;
+                    }
+                    writeStateToStream(stateResponse.getHits().getAt(0).getSourceRef(), restoreStream);
+                } finally {
+                    stateResponse.decRef();
```

Comment:
@droberts195 sorry for another ping, just wanting to double check all these things and that decRef is good and writeStateToStream doesn't asynchronously require a reference